Dense Image Warp tests are flaky

Recently we've seen that #53 is causing flaky failures in the CI. See:
https://source.cloud.google.com/results/invocations/8f31faef-505a-440e-b75f-e6edf1071269/targets/tensorflow_addons%2Fubuntu%2Fgpu%2Fpy3%2Fpresubmit/log

Do you mind taking a look when time allows @WindQAQ ?

@WindQAQ One more is failing that is blocking us to continue building or testing with master:
```
[  FAILED  ] DenseImageWarpTest.test_size_exception
[ RUN      ] DenseImageWarpTest.test_zero_flows
[       OK ] DenseImageWarpTest.test_zero_flows
======================================================================
FAIL: test_size_exception (__main__.DenseImageWarpTest)
test_size_exception (__main__.DenseImageWarpTest)
Make sure it throws an exception for images that are too small.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 37, in testPartExecutor
    yield
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 162, in run
    testMethod()
  File "/root/.cache/bazel/_bazel_root/e051f2f195071208ea7f081f88730f4f/execroot/__main__/bazel-out/k8-opt/bin/tensorflow_addons/image/dense_image_warp_test.runfiles/__main__/tensorflow_addons/image/dense_image_warp_test.py", line 223, in test_size_exception
    self._check_interpolation_correctness(shape, "float32", "float32")
AssertionError: "Grid width must be at least 2." does not match "indices[0,0] = -1 is not in [0, 2)
	 [[{{node dense_image_warp/StatefulPartitionedCall/interpolate_bilinear/gather-top_left/GatherV2}}]] [Op:__inference_dense_image_warp_9536]"
```

@WindQAQ one more is failing now, sorry it was not failing earlier:

```
[  FAILED  ] DenseImageWarpTest.test_size_exception
[ RUN      ] DenseImageWarpTest.test_zero_flows
[       OK ] DenseImageWarpTest.test_zero_flows
======================================================================
FAIL: test_size_exception (__main__.DenseImageWarpTest)
test_size_exception (__main__.DenseImageWarpTest)
Make sure it throws an exception for images that are too small.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 37, in testPartExecutor
    yield
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 162, in run
    testMethod()
  File "/root/.cache/bazel/_bazel_root/e051f2f195071208ea7f081f88730f4f/execroot/__main__/bazel-out/k8-opt/bin/tensorflow_addons/image/dense_image_warp_test.runfiles/__main__/tensorflow_addons/image/dense_image_warp_test.py", line 223, in test_size_exception
    self._check_interpolation_correctness(shape, "float32", "float32")
AssertionError: "Grid width must be at least 2." does not match "indices[0,0] = -1 is not in [0, 2)
	 [[{{node dense_image_warp/StatefulPartitionedCall/interpolate_bilinear/gather-top_left/GatherV2}}]] [Op:__inference_dense_image_warp_9536]"

----------------------------------------------------------------------
Ran 8 tests in 52.030s

FAILED (failures=1)
================================================================================
```

--- a/tensorflow_addons/image/dense_image_warp.py
+++ b/tensorflow_addons/image/dense_image_warp.py
@@ -21,7 +21,6 @@ import numpy as np
 import tensorflow as tf
 
 
-@tf.function
 def interpolate_bilinear(grid, query_points, indexing="ij", name=None):
     """Similar to Matlab's interp2 function.
 
@@ -48,30 +47,32 @@ def interpolate_bilinear(grid, query_points, indexing="ij", name=None):
     with tf.name_scope(name or "interpolate_bilinear"):
         grid = tf.convert_to_tensor(grid)
         query_points = tf.convert_to_tensor(query_points)
-        shape = grid.get_shape().as_list()
-        if len(shape) != 4:
-            msg = "Grid must be 4 dimensional. Received size: "
-            raise ValueError(msg + str(grid.get_shape()))
 
-        batch_size, height, width, channels = (tf.shape(grid)[0],
-                                               tf.shape(grid)[1],
-                                               tf.shape(grid)[2],
-                                               tf.shape(grid)[3])
+        if len(grid.shape) != 4:
+            msg = "Grid must be 4 dimensional. Received size: "
+            raise ValueError(msg + str(grid.shape))
+
+        if len(query_points.shape) != 3:
+            raise ValueError("Query points must be 3 dimensional.")
+
+        grid_shape = tf.shape(grid)
+        query_shape = tf.shape(query_points)
+
+        batch_size, height, width, channels = (grid_shape[0],
+                                               grid_shape[1],
+                                               grid_shape[2],
+                                               grid_shape[3])
 
         shape = [batch_size, height, width, channels]
+        num_queries = query_shape[1]
+
         query_type = query_points.dtype
         grid_type = grid.dtype
 
         tf.debugging.assert_equal(
-            len(query_points.get_shape()),
-            3,
-            message="Query points must be 3 dimensional.")
-        tf.debugging.assert_equal(
-            tf.shape(query_points)[2],
+            query_shape[2],
             2,
             message="Query points must be size 2 in dim 2.")
-
-        num_queries = tf.shape(query_points)[1]
 
         tf.debugging.assert_greater_equal(
             height, 2, message="Grid height must be at least 2."),

--- a/tensorflow_addons/image/dense_image_warp.py
+++ b/tensorflow_addons/image/dense_image_warp.py
@@ -58,10 +58,8 @@ def interpolate_bilinear(grid, query_points, indexing="ij", name=None):
         grid_shape = tf.shape(grid)
         query_shape = tf.shape(query_points)
 
-        batch_size, height, width, channels = (grid_shape[0],
-                                               grid_shape[1],
-                                               grid_shape[2],
-                                               grid_shape[3])
+        batch_size, height, width, channels = (grid_shape[0], grid_shape[1],
+                                               grid_shape[2], grid_shape[3])
 
         shape = [batch_size, height, width, channels]
         num_queries = query_shape[1]
@@ -70,9 +68,7 @@ def interpolate_bilinear(grid, query_points, indexing="ij", name=None):
         grid_type = grid.dtype
 
         tf.debugging.assert_equal(
-            query_shape[2],
-            2,
-            message="Query points must be size 2 in dim 2.")
+            query_shape[2], 2, message="Query points must be size 2 in dim 2.")
 
         tf.debugging.assert_greater_equal(
             height, 2, message="Grid height must be at least 2."),

--- a/tensorflow_addons/image/dense_image_warp_test.py
+++ b/tensorflow_addons/image/dense_image_warp_test.py
@@ -134,7 +134,7 @@ class DenseImageWarpTest(tf.test.TestCase):
 
         self.assertAllClose(rand_image, interp)
 
-    # TODO: run in both graph and eager modes
+    @test_utils.run_in_graph_and_eager_modes
     def test_zero_flows(self):
         """Apply _check_zero_flow_correctness() for a few sizes and types."""
         shapes_to_try = [[3, 4, 5, 6], [1, 2, 2, 1]]

